packages/js-lib/src/zod/zod.shared.schemas.ts

```diff
@@ -54,11 +63,59 @@ function semVer(): z.ZodString {
     .describe('SemVer')
 }
 
-function isoDate(): ZodBrandedString<IsoDate> {
-  return z
-    .string()
-    .regex(/^\d{4}-\d{2}-\d{2}$/, { error: 'Must be a YYYY-MM-DD string' })
-    .describe('IsoDate') as ZodBrandedString<IsoDate>
+export interface CustomZodIsoDateParams {
+  before?: IsoDate
+  sameOrBefore?: IsoDate
+  after?: IsoDate
+  sameOrAfter?: IsoDate
+  between?: { min: IsoDate; max: IsoDate; incl: Inclusiveness }
+}
+
+export interface JsonSchemaDescriptionParams {
+  schema: 'isoDate'
+  params: CustomZodIsoDateParams
+}
+
+function isoDate(params: CustomZodIsoDateParams = {}): ZodBrandedString<IsoDate> {
+  const { before, sameOrBefore, after, sameOrAfter, between } = params
+
+  _assert(Object.keys(params).length <= 1, 'Only one condition is allowed in `isoDate()`!')
+
+  let error = 'Should be be a YYYY-MM-DD string'
+  if (after) error = `Should be after ${after}`
+  if (sameOrAfter) error = `Should be on or after ${sameOrAfter}`
+  if (before) error = `Should be before ${before}`
+  if (sameOrBefore) error = `Should be on or before ${sameOrBefore}`
+  if (between) {
+    const { min, max, incl } = between
+    error = `Should be between ${min} and ${max} (incl: ${incl})`
+  }
+
+  let schema = z.string().refine(
+    dateString => {
+      if (!localDate.isValidString(dateString)) return false
```

Comment:
Just a note that I'm afraid of the performance implications.
In reality we deal with hundreds/thousands of isoDates per "validation run", so, it's very performance-sensitive code, I'd like to scrutinize it.

For example, we could get away with just string comparisons and not relying on LocalDate api.

`isValidString` can be instead done via regex?

Also, I'd like a fast-pass for case when there are no parameters